### PR TITLE
Upgrade examples to core 1.8.1 and upgraded build system

### PR DESCRIPTION
**Pull request by flyingsilverfin**
## What is the goal of this PR?
Upgrade tests and examples to Grakn Core 1.8.1, and the associated changes in the build system. This includes using the centralised maven version declaration, artifact-based grakn distribution for tests, and reordered WORKSPACE file.

## What are the changes implemented in this PR?
* Upgrade deps to be 1.8.1 core compatible
* Upgrade dependencies to use new `@maven` declarations
* Reorder WORKSPACE file
* Use grakn-core artifact for tests

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -87,19 +88,19 @@ jobs:
...
     working_directory: ~/examples
     steps:
-      - install-bazel-linux-rbe
+      - install-bazel
       - checkout
       - run: sudo apt-get update
       - run: sudo apt-get install python3-tk tk-dev  # Required for tkinter
-      - run: pyenv install 3.6.3
-      - run: pyenv global 3.6.3 2.7.12
+#      - run: pyenv install 3.6.3
```
Comment (lolski) [inline .circleci/config.yml]: can we remove the commented code?

File: WORKSPACE [hunk 2]
```diff
@@ -15,54 +15,94 @@
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 #
 
-workspace(name = "graknlabs_examples")
+workspace(name = "graknlabs_examples_core")
```
Comment (lolski) [inline WORKSPACE]: `graknlabs_examples`

File: WORKSPACE [hunk 3]
```diff
@@ -15,54 +15,94 @@
...
+unuseddeps_deps()
+
+# Load Dependencies Kotlin Dependencies
+load("@graknlabs_dependencies//dependencies/maven:artifacts.bzl",
+graknlabs_dependencies_artifacts = "artifacts")
+
+#####################################################################
+# Load @graknlabs_bazel_distribution from (@graknlabs_dependencies) #
+#####################################################################
+#load("@graknlabs_dependencies//dependencies/graknlabs:repositories.bzl", "graknlabs_bazel_distribution")
```
Comment (lolski) [inline WORKSPACE]: can we clean up this comments

File: WORKSPACE [hunk 4]
```diff
@@ -73,115 +113,148 @@ graknlabs_bazel_distribution_pip_install = "pip_install")
...
 
 
-###########################
-# Load Local Dependencies #
-###########################
+load("@graknlabs_bazel_distribution//github:dependencies.bzl", "tcnksm_ghr")
+tcnksm_ghr()
 
-# for Java
+load("@bazel_tools//tools/build_defs/repo:git.bzl", "git_repository")
```
Comment (lolski) [inline WORKSPACE]: do we need skydoc?

File: WORKSPACE [hunk 5]
```diff
@@ -73,115 +113,148 @@ graknlabs_bazel_distribution_pip_install = "pip_install")
...
+bazel_rules_docker_repositories = "repositories")
+bazel_rules_docker_repositories()
 
+load("@io_bazel_rules_docker//repositories:deps.bzl", bazel_rules_docker_container_deps = "deps")
+bazel_rules_docker_container_deps()
 
-##############################
-# Load Protocol Dependencies #
-##############################
+load("@io_bazel_rules_docker//container:container.bzl", "container_pull")
```
Comment (lolski) [inline WORKSPACE]: do we need docker?

File: WORKSPACE [hunk 6]
```diff
@@ -73,115 +113,148 @@ graknlabs_bazel_distribution_pip_install = "pip_install")
...
+    graknlabs_dependencies_artifacts +
+    graknlabs_examples_artifacts,
+)
 
-#####################################
-# Load Bazel Common Workspace Rules #
-#####################################
+###############################################
+# Create @graknlabs_examples_workspace_refs #
+###############################################
```
Comment (lolski) [inline WORKSPACE:224]:
```
#############################################
# Create @graknlabs_examples_workspace_refs #
#############################################
```

**Review by lolski — CHANGES_REQUESTED** [review]

**Review by lolski — APPROVED** [review]

File: WORKSPACE (hunk 4 above)
Comment (flyingsilverfin) [reply WORKSPACE]: nop

**Pull request merged**